apps/server/lib/lexical/server/project/diagnostics.ex

```diff
@@ -168,6 +168,10 @@ defmodule Lexical.Server.Project.Diagnostics do
           )
       end
     end
+
+    defp test_file?(source_file) do
+      String.ends_with?(source_file.path, "_test.exs")
```

Comment:
Small correction/amendment: this also happens with others script files (ending in .exs). `mix compile` compiles all .ex files that are in the given path. `.exs` files are scripts and not compiled but run directly by the elixir interpreter, hence *I think* it's better to change this check. For example Phoenix/Ecto also use .exs files for the migration definitions (usually found in priv/repo/migrations)

Reply:
Thanks for the suggestions, will change this later